### ISSUE TITLE
Use hex-serialization for byte-array based types

### ISSUE THREAD

**Issue by Voxelot** [FuelLabs/fuel-types#31]
Working with fuel-types would be much more convenient if serde used hex serialization for `key!` types instead of the default behavior of stringifying an array of ints (`[1,2,3,4,..]`). 

We already use hex encoding for [Display](https://github.com/FuelLabs/fuel-types/blob/master/src/types.rs#L222) and [Debug](https://github.com/FuelLabs/fuel-types/blob/master/src/types.rs#L216), we should do the same for serde.

**Comment by kayagokalp** [comment]
A quick question, while I am trying to test I got confused about some parts.  After #28, it seems like we longer have the `serde-types-minimal` feature but `test_key_with_big_array` test in [types.rs](https://github.com/FuelLabs/fuel-types/blob/c9048441ce982f4e0bdf94e5e39f3fc7560bfc4e/tests/types.rs#L131) (also the same configuration flag is used in the implementation as well) depends on it. 

Also looking into CI for #28, for `cargo-verifications (check, --all-targets --all-features)` job, I can see that `test_key_with_big_array` test did not run. Is this the desired behavior?

**Comment by Voxelot** [comment]
This is a bug and should be fixed.

**Comment by kayagokalp** [comment]
To test current hex serialization/de-serialization with `serde`, I am currently adding `serde-types-minimal = ["serde"]` under `[features]` in `Cargo.toml`. Should I open a new issue for this one?

**Comment by Voxelot** [comment]
Yes a separate issue would make sense.
